### PR TITLE
refactoring: converting to the same type of scripts

### PR DESCRIPTION
**Pull request by schnek**
**Description**
added macros to reduce string length …
 - #define MAP_GET_CREATURE_NEAREST_COORDS GetMapMgr()->GetInterface()->GetCreatureNearestCoords

ps:
- looks like it macro MAP_GET_CREATURE_NEAREST_COORDS  -> getNearestCreature();
- scripts where in the message .c_str() it was not possible to test (before the changes did not work either)

**Todo / Checklist**
- [x] Target is branch develop.
- [x] Detailed description about this pull request.
- [x] Checked AE-Coding standards.

**Tests Performed:** 
- [x] Build AE.
- [x] Server startup.
- [x] Log into world.

**Multiversion Ingame Tests Performed:**
- [x] WotLK
- [x] Cata

### CODE REVIEW

File: src/world/Macros/ScriptMacros.hpp [hunk 1]
```diff
@@ -5,6 +5,12 @@ This file is released under the MIT license. See README-MIT for more information
 
 #pragma once
 
+/// Coords Creature
+#define MAP_CREATURE_NEAREST_COORDS GetMapMgr()->GetInterface()->GetCreatureNearestCoords
```
Comment (Appled) [inline src/world/Macros/ScriptMacros.hpp:9]: Rename to MAP_GET_CREATURE_NEAREST_COORDS

File: src/world/Macros/ScriptMacros.hpp [hunk 2]
```diff
@@ -5,6 +5,12 @@ This file is released under the MIT license. See README-MIT for more information
...
+
+/// Coords GameObject
+#define MAP_GAMEOBJECT_NEAREST_COORDS GetMapMgr()->GetInterface()->GetGameObjectNearestCoords
```
Comment (Appled) [inline src/world/Macros/ScriptMacros.hpp:12]: Rename to MAP_GET_GAMEOBJECT_NEAREST_COORDS

File: src/world/Map/MapCell.h [hunk 3]
```diff
@@ -49,8 +49,10 @@ friend class MapMgr;
     void QueueUnloadPending();
     void CancelPendingUnload();
     void Unload();
+    /// xyz
```
Comment (Appled) [inline src/world/Map/MapCell.h:52]: No need to add this

File: src/world/Map/MapCell.h [hunk 4]
```diff
@@ -49,8 +49,10 @@ friend class MapMgr;
     void QueueUnloadPending();
     void CancelPendingUnload();
     void Unload();
+    /// xyz 
     inline uint16_t GetPositionX() { return _x; }
     inline uint16_t GetPositionY() { return _y; }
+    // inline uint16_t GetPositionZ() { return _z; }
```
Comment (Appled) [inline src/world/Map/MapCell.h:55]: No need to add this

File: src/world/Map/MapCell.h [hunk 5]
```diff
@@ -62,6 +64,7 @@ friend class MapMgr;
 private:
     uint16_t _x;
     uint16_t _y;
+    // uint16_t _z;
```
Comment (Appled) [inline src/world/Map/MapCell.h:67]: No need to add this

File: src/world/Macros/ScriptMacros.hpp (hunk 1 above)
Comment (schnek) [reply src/world/Macros/ScriptMacros.hpp:9]: @Appled ok. about the method they are equal 

> MAP_GET_CREATURE_NEAREST_COORDS = getNearestCreature();

Comment (schnek) [reply src/world/Macros/ScriptMacros.hpp]: look here =)

**Comment by Zyres** [comment]
Makros are evil, they are a headache to debug and we should not introduce functional makros.
If there is a need to reduce the length of code, choose to create a function in class Object or Unit.

**Comment by schnek** [comment]
```
Makros are evil, they are a headache to debug and we should not introduce functional makros.
If there is a need to reduce the length of code, choose to create a function in class Object or Unit.
```
yes, macros are evil and deleting all of them would be a great victory.


this function seems to be --> https://github.com/AscEmu/AscEmu/pull/926/files#diff-2354b11f212479564c7fad61d8a6a5c5aa3a3e521ad4d89232261ee41e1a95d1L94

**Pull request closed without merging**